### PR TITLE
Update command - slcli object-storage endpoints

### PR DESCRIPTION
**Pull request by edsonarios**
Issue link: #1682 
```
softlayer-python/ (issue1682 *) $ slcli object-storage endpoints 123456
┌────────┬───────────────┬────────────────┬─────────────────┬──────────────────────────────────────────────────────────┐
│ Legacy │ EndPoint Type │ Public/Private │ Location/Region │                           Url                            │
├────────┼───────────────┼────────────────┼─────────────────┼──────────────────────────────────────────────────────────┤
│ False  │ Cross Region  │     Public     │     us-geo      │        s3.us.cloud-object-storage.appdomain.cloud        │
│ False  │ Cross Region  │     Public     │     Dallas      │      s3.dal.us.cloud-object-storage.appdomain.cloud      │
│ False  │ Cross Region  │     Public     │   Washington    │      s3.wdc.us.cloud-object-storage.appdomain.cloud      │
│ False  │ Cross Region  │     Public     │    San Jose     │      s3.sjc.us.cloud-object-storage.appdomain.cloud      │
│ False  │ Cross Region  │     Public     │     eu-geo      │        s3.eu.cloud-object-storage.appdomain.cloud        │
│ False  │ Cross Region  │     Public     │    Amsterdam    │      s3.ams.eu.cloud-object-storage.appdomain.cloud      │
│ False  │ Cross Region  │     Public     │    Frankfurt    │      s3.fra.eu.cloud-object-storage.appdomain.cloud      │
│ False  │ Cross Region  │     Public     │      Milan      │      s3.mil.eu.cloud-object-storage.appdomain.cloud      │
│ False  │ Cross Region  │     Public     │     ap-geo      │        s3.ap.cloud-object-storage.appdomain.cloud        │
│ False  │ Cross Region  │     Public     │      Tokyo      │      s3.tok.ap.cloud-object-storage.appdomain.cloud      │
│ False  │ Cross Region  │     Public     │      Seoul      │      s3.seo.ap.cloud-object-storage.appdomain.cloud      │
│ False  │ Cross Region  │     Public     │    Hong Kong    │      s3.hkg.ap.cloud-object-storage.appdomain.cloud      │
│ False  │ Cross Region  │    Private     │     us-geo      │    s3.private.us.cloud-object-storage.appdomain.cloud    │
│ False  │ Cross Region  │    Private     │     Dallas      │  s3.private.dal.us.cloud-object-storage.appdomain.cloud  │
│ False  │ Cross Region  │    Private     │   Washington    │  s3.private.wdc.us.cloud-object-storage.appdomain.cloud  │
│ False  │ Cross Region  │    Private     │    San Jose     │  s3.private.sjc.us.cloud-object-storage.appdomain.cloud  │
│ False  │ Cross Region  │    Private     │     eu-geo      │    s3.private.eu.cloud-object-storage.appdomain.cloud    │
│ False  │ Cross Region  │    Private     │    Amsterdam    │  s3.private.ams.eu.cloud-object-storage.appdomain.cloud  │
│ False  │ Cross Region  │    Private     │    Frankfurt    │  s3.private.fra.eu.cloud-object-storage.appdomain.cloud  │
```

### CODE REVIEW

File: SoftLayer/CLI/object_storage/list_endpoints.py [hunk 1]
```diff
@@ -2,26 +2,83 @@
...
     """List object storage endpoints."""
 
     mgr = SoftLayer.ObjectStorageManager(env.client)
-    endpoints = mgr.list_endpoints()
+    endpoints = mgr.list_endpoints(identifier)
+
+    final_end_points = []
 
-    table = formatting.Table(['datacenter', 'public', 'private'])
+    table = formatting.Table(['Legacy', 'EndPoint Type', 'Public/Private', 'Location/Region', 'Url'])
```
Comment (allmightyspiff) [inline SoftLayer/CLI/object_storage/list_endpoints.py:22]: I'd prefer the order be as follows:

```suggestion
    table = formatting.Table(['Location/Region', 'Url', 'EndPoint Type', 'Public/Private', 'Legacy'])
```

Also, set the alignment for `location` and `url` to be left as that makes it read a little easier.

File: SoftLayer/CLI/object_storage/list_endpoints.py [hunk 2]
```diff
@@ -2,26 +2,83 @@
...
     for endpoint in endpoints:
-        table.add_row([
-            endpoint['datacenter']['name'],
-            endpoint['public'],
-            endpoint['private'],
-        ])
+        data = [endpoint['legacy'], end_point_return(endpoint['region']), public_private(endpoint['type']),
+                location_region(endpoint), endpoint['url']]
```
Comment (allmightyspiff) [inline SoftLayer/CLI/object_storage/list_endpoints.py:25]: Setting up data as a list of datapoints makes this code hard to follow. My suggestion would be to have data be a dictionary, and `final_end_points` be a list of those dictionaries. 

`add_array_to_table` would take that list of dictionaries, and return a list of lists that can be added to the table.


```python

data = {
    "legacy": endpoint['legacy'],
    "type": end_point_return(endpoint['region']),
    "publicPrivate": public_private(endpoint['type']),
    "region": location_region(endpoint)
    "url": endpoint['url']
}
final_end_points.append(data)
```

then in `add_array_to_table` and `sort_endpoint` you can do something like `array['legacy']` instead of `array[0]`

That way the code should be a bit easier to follow.
Thanks

**Review by allmightyspiff — CHANGES_REQUESTED** [review]

File: SoftLayer/CLI/object_storage/list_endpoints.py (hunk 1 above)
Comment (edsonarios) [reply SoftLayer/CLI/object_storage/list_endpoints.py]: Done

**Review by allmightyspiff — APPROVED** [review]
